music21/stream/base.py

```diff
@@ -231,6 +231,7 @@ def __init__(self, givenElements=None, *args, **keywords):
             # TODO: perhaps convert a single element into a list?
             for e in givenElements:
                 try:
+                    self.coreGuardBeforeAddElement(e)
```

Comment:
Thanks -- I think though this is a good chance to change the behavior of the Stream constructor when given a list of elements.  My sense is that we check first if all elements have offset 0 then we use coreAppend instead of coreInsert.  The though originally was that we would do things like:

```
>>> n1 = note.Note('C4')
>>> n1.offset = 0.0
>>> n2 = note.Note('D4')
>>> n2.offset = 1.0
>>> s = stream.Stream([n1, n2])
```

but since about v1 of music21, we've gotten away from setting the offsets of notes etc. before adding to Streams.  A lot of our functions though still assume we'll do that.  Instead more often people want to do this:

```
>>> s = stream.Stream([note.Note('C4'), note.Note('D4')])
```

and they're shocked that both notes are at offset 0.

So a test that if all offsets are 0 we append, otherwise we insert. -- [but maybe a check that it's not a list of parts??? since those SHOULD all be at zero]

This is also definitely the time to change the behavior of line 230, to check that givenElements is not a single Stream object.  My students always make this error, which should work:

```
>>> n = note.Note()
>>> m = stream.Measure(n)
>>> p = stream.Part(m)
>>> sc = stream.Score(p)
```

and they're shocked that `sc` just contains a single note.  This is a paradigm that we'd like to have work

Reply:
I can see the value in that. My impulse is to extend the all-are-parts rule to be all-are-streams-not-measures to get Voice, Score, Opus, Stream, etc. Only Measure is the supported positioned stream in a stream.

What about clefs, meters, etc. (corner case) -- okay with those being appended if they're listed in the wrong order?

Reply:
well, voice is a supported positioned stream in a stream also, but you can put two voices at 0.0 and we shouldn't mess with that.